app/models/user.rb

```diff
@@ -55,12 +49,8 @@ def friend_requests
   end
 
   def confirm_friend(user)
-    friendship = inverse_friendships.find_by(user_id: user.id)
-    friendship.status = true
-    friendship.save!
-  end
-
-  def friend?(user)
-    friends.include?(user)
+    inverse_friendship = inverse_friendships.find_by(user_id: user.id)
+    inverse_friendship.status = true
+    inverse_friendship.save
   end
```

Comment:
Refer to these links to [this artcile](https://explainextended.com/2009/03/07/selecting-friends) and [stackoverflow discussion](https://stackoverflow.com/questions/4219979/sql-best-practice-for-a-friendship-table) as originally provided for you.

 - [ ] When friendship is confirmed, an additional row with reversed user-friend pair is created. _Are you still using codes from version 1?_